apps/src/netsim/NetSimLobby.js

```diff
@@ -215,86 +273,149 @@ NetSimLobby.prototype.refreshShardList_ = function () {
 
-  this.periodicRefresh_.setActionInterval(isInLobby ?
-      AUTO_REFRESH_INTERVAL_MS : CLOSED_REFRESH_INTERVAL_MS);
+  // Share link state?
 
-  if (isInLobby) {
-    // Show the lobby and connection selector
-    $(this.lobbyOpenDiv_).show();
-    $(this.lobbyClosedDiv_).hide();
+  // Disconnect button state?
 
-    if (!this.connection_.isConnectedToShard()) {
-      $(lobbyList).empty();
-      $(this.connectButton_).hide();
-      return;
-    }
+  this.periodicRefresh_.setActionInterval(CLOSED_REFRESH_INTERVAL_MS);
+  this.periodicRefresh_.enable();
+};
 
-    this.connection_.getAllNodes(function (lobbyData) {
-      $(lobbyList).empty();
-      $(self.connectButton_).show();
+/**
+ * Show preconnect controls (name, shard-select) and actual lobby listing.
+ * @private
+ */
+NetSimLobby.prototype.refreshOpenLobby_ = function () {
+  this.openRoot_.show();
+  this.closedRoot_.hide();
 
-      lobbyData.sort(function (a, b) {
-        if (a.getDisplayName() > b.getDisplayName()) {
-          return 1;
-        }
-        return -1;
-      });
-
-      self.selectedListItem_ = undefined;
-      lobbyData.forEach(function (simNode) {
-        var item = $('<li>').html(
-            simNode.getDisplayName() + ' : ' +
-            simNode.getStatus() + ' ' +
-            simNode.getStatusDetail());
-
-        // Style rows by row type.
-        if (simNode.getNodeType() === NetSimNodeRouter.getNodeType()) {
-          item.addClass('router_row');
-        } else {
-          item.addClass('user_row');
-          if (simNode.entityID === self.connection_.myNode.entityID) {
-            item.addClass('own_row');
-          }
-        }
+  // Do we have a name yet?
+  if (this.nameInput_.val() === '') {
+    this.nameInput_.prop('disabled', false);
+    this.setNameButton_.show();
 
-        // Preserve selected item across refresh.
-        if (simNode.entityID === self.selectedID_) {
-          item.addClass('selected_row');
-          self.selectedListItem_ = item;
-        }
+    this.shardView_.hide();
+    this.periodicRefresh_.disable();
+    return;
+  }
 
-        dom.addClickTouchEvent(item[0], self.onRowClick_.bind(self, item, simNode));
-        item.appendTo(lobbyList);
-      });
+  // We have a name
+  this.nameInput_.prop('disabled', true);
+  this.setNameButton_.hide();
+  this.shardView_.show();
+
+  // Do we have a shard yet?
+  if (!this.connection_.isConnectedToShard()) {
+    this.shardSelector_.val('__none');
+    this.addRouterButton_.hide();
+    this.lobbyList_.hide();
+    this.connectButton_.hide();
+    this.periodicRefresh_.disable();
+    return;
+  }
 
-      self.onSelectionChange();
+  // We have a shard
+  this.addRouterButton_.show();
+  this.lobbyList_.show();
+  this.connectButton_.show();
+  this.refreshLobbyList_();
+};
 
-      self.periodicRefresh_.enable();
+/**
+ * Reload the lobby listing of nodes.
+ * @private
+ */
+NetSimLobby.prototype.refreshLobbyList_ = function () {
+  var self = this;
```

Comment:
Lately I've been binding functions to this instead of using self. I'm not sure why I didn't in the past - I think I just didn't realize how easy it was. How you have it was fine, but another option would be:
```
this.connection_.getAllNodes(function (lobbydata) {
// i can use this.foo inside here
...
}.bind(this)